### DISCUSSION TITLE
Per-file access reports?

### DISCUSSION THREAD

**Discussion by imjasonh** [awslabs/soci-snapshotter#1182] [Q&A]
I'd like to see if I can use SOCI and kind/k3s to run end-to-end tests, and get a report about how much of the image was/wasn't used during tests. This could be helpful for both identifying unnecessary files, and for improving test coverage if I expect certain files to be accessed during tests.

I looked through docs and issues and couldn't find any information about per-file access logs, but it's possible I missed something.

If this is possible and interesting, but just not implemented yet, I'd be willing to contribute changes to enable this.

**Comment by Kern--** [comment]
You might be able to cobble this together from the fuse logs, but I think it would also be interesting to produce metrics or something to make it easier to get insights. 

For access logs, you could add this to your config:
```
[fuse]
log_fuse_operations=true
```

And you'll get a log line for each operation performed on each file. 

We also have more verbose logging, by setting
```
debug=true
```
and passing `--log-level trace` when launching the snapshotter. This will give the raw FUSE transactions which gives you a little more info, but also requires more processing (since the operations are done by inode rather than path, so you'd have to reconstruct paths).
